diff -r -u --strip-trailing-cr hdf-4.2.12.original/config/cmake_ext_mod/HDFMacros.cmake hdf-4.2.12/config/cmake_ext_mod/HDFMacros.cmake
--- hdf-4.2.12.original/config/cmake_ext_mod/HDFMacros.cmake	2017-05-17 07:50:42.265738600 +0200
+++ hdf-4.2.12/config/cmake_ext_mod/HDFMacros.cmake	2017-05-17 07:50:40.630158300 +0200
@@ -73,22 +73,25 @@
 
 #-------------------------------------------------------------------------------
 macro (HDF_SET_LIB_OPTIONS libtarget libname libtype)
+  if (MINGW)
+    string(REGEX REPLACE "^lib([A-za-z]+)" "\\1" libname ${libname})
+  endif () 
   if (${libtype} MATCHES "SHARED")
-    if (WIN32)
+    if (MSVC)
       set (LIB_RELEASE_NAME "${libname}")
       set (LIB_DEBUG_NAME "${libname}_D")
-    else (WIN32)
+    else (MSVC)
       set (LIB_RELEASE_NAME "${libname}")
       set (LIB_DEBUG_NAME "${libname}_debug")
-    endif (WIN32)
+    endif (MSVC)
   else (${libtype} MATCHES "SHARED")
-    if (WIN32)
+    if (MSVC)
       set (LIB_RELEASE_NAME "lib${libname}")
       set (LIB_DEBUG_NAME "lib${libname}_D")
-    else (WIN32)
+    else (MSVC)
       set (LIB_RELEASE_NAME "${libname}")
       set (LIB_DEBUG_NAME "${libname}_debug")
-    endif (WIN32)
+    endif (MSVC)
   endif (${libtype} MATCHES "SHARED")
 
   set_target_properties (${libtarget}
@@ -99,15 +102,21 @@
       OUTPUT_NAME_RELWITHDEBINFO ${LIB_RELEASE_NAME}
   )
 
-  #----- Use MSVC Naming conventions for Shared Libraries
-  if (MINGW AND ${libtype} MATCHES "SHARED")
-    set_target_properties (${libtarget}
-        PROPERTIES
-        IMPORT_SUFFIX ".lib"
-        IMPORT_PREFIX ""
-        PREFIX ""
-    )
-  endif (MINGW AND ${libtype} MATCHES "SHARED")
+  ####----- Use MSVC Naming conventions for Shared Libraries
+  ###if (MINGW AND ${libtype} MATCHES "SHARED")
+  ###  set_target_properties (${libtarget}
+  ###      PROPERTIES
+  ###      IMPORT_SUFFIX ".dll.a"
+  ###      IMPORT_PREFIX "lib"
+  ###      PREFIX ""
+  ###  )
+  ###endif (MINGW AND ${libtype} MATCHES "SHARED")
+  if (MINGW)
+    set_target_properties (${libtarget} PROPERTIES OUTPUT_NAME ${libname} ARCHIVE_OUTPUT_NAME ${libname})
+    if (${libtype} MATCHES "SHARED")
+      set_target_properties (${libtarget} PROPERTIES RUNTIME_OUTPUT_NAME ${libname})
+    endif ()
+  endif (MINGW)
 
 endmacro (HDF_SET_LIB_OPTIONS)
 
@@ -126,17 +135,17 @@
 
   if (${libtype} MATCHES "SHARED")
     if (WIN32)
-      if (MINGW)
+      if (MSVC)
         set_target_properties (${libtarget} PROPERTIES
             IMPORTED_IMPLIB "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${IMPORT_LIB_NAME}.lib"
             IMPORTED_LOCATION "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${IMPORT_LIB_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX}"
         )
-      else (MINGW)
+      else (MSVC)
         set_target_properties (${libtarget} PROPERTIES
             IMPORTED_IMPLIB "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_BUILD_TYPE}/${CMAKE_IMPORT_LIBRARY_PREFIX}${IMPORT_LIB_NAME}${CMAKE_IMPORT_LIBRARY_SUFFIX}"
             IMPORTED_LOCATION "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_BUILD_TYPE}/${CMAKE_IMPORT_LIBRARY_PREFIX}${IMPORT_LIB_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX}"
          )
-      endif (MINGW)
+      endif (MSVC)
     else (WIN32)
       if (CYGWIN)
         set_target_properties (${libtarget} PROPERTIES
