--- hdf5-1.8.19.orig/src/H5win32defs.h	2019-02-16 23:41:12.000000000 +0100
+++ hdf5-1.8.19/src/H5win32defs.h	2019-03-30 08:41:48.010078700 +0100
@@ -54,7 +54,7 @@
  * Also note that the variadic macro is using a VC++ extension
  * where the comma is dropped if nothing is passed to the ellipsis.
  */
-#define HDopen(S,F,...)       _open(S, F | _O_BINARY, __VA_ARGS__)
+#define HDopen(S,F,...)       _open(S, F | _O_BINARY, ##__VA_ARGS__)
 #define HDread(F,M,Z)       _read(F,M,Z)
 #define HDrmdir(S)          _rmdir(S)
 #define HDsetvbuf(F,S,M,Z)  setvbuf(F,S,M,(Z>1?Z:2))
